src/plugins/data/public/ui/query_string_input/language_switcher.test.tsx

```diff
@@ -55,4 +55,32 @@ describe('LanguageSwitcher', () => {
     expect(component.find(EuiPopover).prop('isOpen')).toBe(true);
     expect(component.find('[data-test-subj="languageToggle"]').get(0).props.checked).toBeTruthy();
   });
+
+  describe('for eql', () => {
+    it('should show radio button for lucene, kql and eql', () => {
+      const component = mountWithIntl(
+        wrapInContext({
+          language: 'kuery',
+          onSelectLanguage: () => {
+            return;
+          },
+          includeEqlLanguage: true,
+        })
+      );
+      component.find(EuiButtonEmpty).simulate('click');
+      expect(component.find(EuiPopover).prop('isOpen')).toBe(true);
```

Comment:
nit: Since this isn't a Cypress test (where we might implicitly be waiting for a UI state update), is this `expect` necessary? Would it be reasonable to split this into a series of tests with one `expect` per test?